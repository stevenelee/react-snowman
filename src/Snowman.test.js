import { render, fireEvent } from "@testing-library/react";
import Snowman from './Snowman.js';
import img0 from "./0.png";
import img1 from "./1.png";
import img2 from "./2.png";
import img3 from "./3.png";
import img4 from "./4.png";
import img5 from "./5.png";
import img6 from "./6.png";


it("losing functionality is working", function () {
  const { container } = render(
      <Snowman
          images={[img0, img1, img2, img3, img4, img5, img6]}
          words="testword"
          maxWrong={6}
      />
      );

  // simulate losing
  const bKey = container.querySelector('[value="b"]');
  fireEvent.click(bKey);

  const dKey = container.querySelector('[value="d"]');
  fireEvent.click(dKey);

  const fKey = container.querySelector('[value="f"]');
  fireEvent.click(fKey);

  const gKey = container.querySelector('[value="g"]');
  fireEvent.click(gKey);

  const hKey = container.querySelector('[value="h"]');
  fireEvent.click(hKey);

  const iKey = container.querySelector('[value="i"]');
  fireEvent.click(iKey);

  // snapshot test when losing
  expect(container).toMatchSnapshot();

  // no images should appear after losing
  const jKey = container.querySelector('[value="j"]');
  fireEvent.click(jKey);

  expect(container.querySelector('img[alt="7"]')).not.toBeInTheDocument();

})