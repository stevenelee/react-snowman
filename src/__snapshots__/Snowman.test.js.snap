// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`losing functionality is working 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p
      class="Snowman-wrong"
    >
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
    </p>
    <p>
      YOU LOSE
      <br />
      Word is: 
      t
    </p>
  </div>
</div>
`;
